californium-core/src/main/java/org/eclipse/californium/core/network/MessageIdTracker.java

```diff
@@ -57,7 +57,7 @@ public MessageIdTracker(final NetworkConfig config) {
 		exchangeLifetime = config.getLong(NetworkConfig.Keys.EXCHANGE_LIFETIME);
 		boolean useRandomFirstMID = config.getBoolean(NetworkConfig.Keys.USE_RANDOM_MID_START);
 		if (useRandomFirstMID) {
-			counter = new AtomicInteger(new Random().nextInt(1 << 10));
+			counter = new AtomicInteger(new SecureRandom().nextInt(1 << 10));
```

Comment:
There is no need to use a cryptographically strong random in order to create the initial counter value. The counter will be incremented sequentially only afterwards ...